addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -804,9 +804,12 @@ options.registry.WebsiteFormEditor = FormEditor.extend({
         if (formInfo) {
             const formatInfo = this._getDefaultFormat();
             await formInfo.formFields.forEach(async field => {
-                field.formatInfo = formatInfo;
-                await this._fetchFieldRecords(field);
-                this.$target.find('.s_website_form_submit, .s_website_form_recaptcha').first().before(this._renderField(field));
+                // Create a shallow copy to prevent unintended mutations to the
+                // original field stored in FormRegistry.
+                const _field = {...field};
+                _field.formatInfo = formatInfo;
+                await this._fetchFieldRecords(_field);
+                this.$target.find(".s_website_form_submit, .s_website_form_recaptcha").first().before(this._renderField(_field));
```

Comment:
I would use vanilla JS by access this$target[0]

Reply:
Thanks, I'll do that.